### PR TITLE
[cloud_firestore] Invoke on ui thread only

### PR DESCRIPTION
**Pull request by kroikie**
## Description

In coming versions of the Flutter engine calls to the Dart side from the Java side must be done on the UI thread. The Firestore plugin implements transactions by making calls to the Dart side from the Java side. This change ensures that those calls are made on the UI thread.

## Related Issues

Resolves: https://github.com/flutter/flutter/issues/32657

### CODE REVIEW

File: packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/FlutterFirebaseAppRegistrar.java [hunk 1]
```diff
@@ -7,8 +7,8 @@
 import java.util.List;
 
 public class FlutterFirebaseAppRegistrar implements ComponentRegistrar {
-  private static final String LIBRARY_NAME = "flutter-firebase_cloud_firestore";
-  private static final String LIBRARY_VERSION = "0.12.1";
+  private static final String LIBRARY_NAME = "flutter-fire-fst";
```
Comment (collinjackson) [inline packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/FlutterFirebaseAppRegistrar.java]: Should be mentioned in changelog

File: packages/cloud_firestore/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## 0.12.2
+
+* Ensure that all channel calls to the Dart side from the Java side are done
```
Comment (collinjackson) [inline packages/cloud_firestore/CHANGELOG.md]: You may want to mention that this is an Android-only change.

File: packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/FlutterFirebaseAppRegistrar.java (hunk 1 above)
Comment (kroikie) [reply packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/FlutterFirebaseAppRegistrar.java]: Done

File: packages/cloud_firestore/CHANGELOG.md (hunk 2 above)
Comment (kroikie) [reply packages/cloud_firestore/CHANGELOG.md]: Done

File: packages/cloud_firestore/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
...
+  on the UI thread. This change allows Transactions to work with upcoming
+  Engine restrictions, which require channel calls be made on the UI thread. 
+
```
Comment (larssn) [inline packages/cloud_firestore/CHANGELOG.md]: Does this take into account that Firestore's Android SDK will not allow transactions on the UI thread?

Comment (kroikie) [reply packages/cloud_firestore/CHANGELOG.md]: The transaction itself is not being executed on the UI thread, only the message being passed over the channel from native side to Dart side is being done on the UI thread.

**Comment by ride4sun** [comment]
This does not fix flutter #18870

**Pull request merged**